driver/generic/ili9xxx.py

```diff
@@ -0,0 +1,148 @@
+_MADCTL_MV = const(0x20)  # page/column order (0: normal mode 1; reverse mode)
+_MADCTL_ML = const(
+    0x10
+)  # line address order (0: refresh to to bottom; 1: refresh bottom to top)
+_MADCTL_BGR = const(0x08)  # colors are BGR (not RGB)
+_MADCTL_RTL = const(0x04)  # refresh right to left
+
+_MADCTL_ROTS = (
+    const(_MADCTL_MX),  # 0 = portrait
+    const(_MADCTL_MV),  # 1 = landscape
+    const(_MADCTL_MY),  # 2 = inverted portrait
+    const(_MADCTL_MX | _MADCTL_MY | _MADCTL_MV),  # 3 = inverted landscape
+)
+
+
+class Ili9341_hw(St77xx_hw):
+    def __init__(self, res, **kw):
+        super().__init__(
+            res=res,
+            suppRes=[
+                (240, 320),
+            ],
+            model=None,
+            suppModel=None,
+            bgr=True,
+            **kw,
+        )
+
+    def config_hw(self):
+        self._run_seq(
+            [
+                (_SLPOUT, None, 100),
+                (_PWCTRB, b"\x00\xC1\x30"),  # Pwr ctrl B
+                (_POSC, b"\x64\x03\x12\x81"),  # Pwr on seq. ctrl
+                (_DTCA, b"\x85\x00\x78"),  # Driver timing ctrl A
+                (_PWCTRA, b"\x39\x2C\x00\x34\x02"),  # Pwr ctrl A
+                (_PUMPRC, b"\x20"),  # Pump ratio control
+                (_DTCB, b"\x00\x00"),  # Driver timing ctrl B
+                (_PWCTR1, b"\x23"),  # Pwr ctrl 1
+                (_PWCTR2, b"\x10"),  # Pwr ctrl 2
+                (_VMCTR1, b"\x3E\x28"),  # VCOM ctrl 1
+                (_VMCTR2, b"\x86"),  # VCOM ctrl 2
+                # (_MADCTL, self.rotation"),  # Memory access ctrl  # TODO: rotation?
+                (_VSCRSADD, b"\x00"),  # Vertical scrolling start address
+                (_PIXFMT, b"\x55"),  # COLMOD: Pixel format
+                (_FRMCTR1, b"\x00\x18"),  # Frame rate ctrl
+                (_DFUNCTR, b"\x08\x82\x27"),
+                (_ENABLE3G, b"\x00"),  # Enable 3 gamma ctrl
+                (_GAMMASET, b"\x01"),  # Gamma curve selected
+                (
+                    _GMCTRP1,
+                    b"\x0F\x31\x2B\x0C\x0E\x08\x4E\xF1\x37\x07\x10\x03\x0E\x09\x00",
+                ),
+                (
+                    _GMCTRN1,
+                    b"\x00\x0E\x14\x03\x11\x07\x31\xC1\x48\x08\x0F\x0C\x31\x36\x0F",
+                ),
+                (_SLPOUT, None, 100),
+                (_DISPLAY_ON, None),
+            ]
+        )
+
+    def apply_rotation(self, rot):
+        self.rot = rot
+        if (self.rot % 2) == 0:
+            self.width, self.height = self.res
+        else:
+            self.height, self.width = self.res
+        self.write_register(
+            _MADCTL,
+            bytes([(_MADCTL_BGR if self.bgr else 0) | _MADCTL_ROTS[self.rot % 4]]),
+        )
+
+
+class Ili9341(Ili9341_hw, St77xx_lvgl):
+    def __init__(self, res, doublebuffer=True, factor=4, **kw):
+        """See :obj:`Ili9341_hw` for the meaning of the parameters."""
+        import lvgl as lv
+
+        Ili9341_hw.__init__(self, res=res, **kw)
```

Comment:
I just copied the style of St7789 which does this, I can remove the res parameter and just hardcode it into Ili9341_hw class.

Reply:
In cases where a display comes in only one flavour with predefined resolution, like ili9341, I don't think it's useful to pass the resolution at all. Hard coding it makes sense.  

This is also true for St7735 and St7789.  
I'm not sure what was the motivation for asking the user to provide a resolution for these displays, when only one is supported on each. (@eudoxos?)

Reply:
St7789 supports quite a few different resolutions, and those are only the ones I laid my hands on during testing. St7735 has only one resolution (so it could be perhaps the default value to the ctor) but I assumed it would also be used with different panels. Datasheets of both chips are generic WRT resolution.

https://github.com/lvgl/lv_binding_micropython/blob/a69f4b514e426a129fc3e891e0bf22a5f88845a9/driver/generic/st77xx.py#L156-L164

Reply:
ok, this makes sense; i'll hard program in the (240, 320), since i believe ili9341 only supports that resolution.